platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/v2/extensions/event/ParameterizedLogMessage.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.v2.extensions.event;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A log message that can be parameterized with arguments. The arguments are inserted into the message at the positions
+ * of the {} placeholders.
+ * <p>
+ * The implementation is copied from slf4j for first tests. need to be replaced in future. SLF4J is using MIT license
+ * (https://github.com/qos-ch/slf4j/blob/master/LICENSE.txt). Based on that we can use it in our project for now
+ *
+ * @param messagePattern the message pattern
+ * @param args           the arguments
+ * @see LogMessage
+ */
+public record ParameterizedLogMessage(@NonNull String messagePattern, @NonNull Object... args) implements LogMessage {
+
+    static final char DELIM_START = '{';
+    static final String DELIM_STR = "{}";
+    private static final char ESCAPE_CHAR = '\\';
+
+    @Override
+    public String getMessage() {
```

Comment:
In my mind, a getter should not be doing so much work, seems counterintuitive

Reply:
Well, it is mentioned in the java doc of the interface that calling the method can create some workload. java.util.function.Supplier uses get() as function name, too. Based on that I would disagree here. 
But: That need to be changed in future anyway since the value need to be cached and should not calculated every time the message is called. Based on that it might look more like a getter in future.